File: backend/models/Event.js
import mongoose from 'mongoose';

const eventSchema = mongoose.Schema(
    {
        title: {
            type: String,
            required: true,
        },
        description: {
            type: String,
            required: true,
        },
        date: {
            type: Date,
            required: true,
        },
        time: {
            type: String,
            required: true,
        },
        location: {
            type: String,
            required: true,
        },
        capacity: {
            type: Number,
            required: true,
        },
        category: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Category',
            required: true,
        },
        organizer: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User',
            required: true,
        },
        price: {
            type: Number,
            default: 0,
        },
    },
    {
        timestamps: true,
    }
);

const Event = mongoose.model('Event', eventSchema);

export default Event;

File: frontend/lib/models/event.dart
class Event {
  final String id;
  final String title;
  final String description;
  final String date;
  final String time;
  final String location;
  final int capacity;
  final String categoryName;
  final String organizerName;
  final double price;
  final int registrationCount;

  Event({
    required this.id,
    required this.title,
    required this.description,
    required this.date,
    required this.time,
    required this.location,
    required this.capacity,
    required this.categoryName,
    required this.organizerName,
    this.price = 0.0,
    this.registrationCount = 0,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['_id'] ?? '',
      title: json['title'] ?? 'No Title',
      description: json['description'] ?? '',
      date: json['date'] ?? DateTime.now().toIso8601String(),
      time: json['time'] ?? '00:00',
      location: json['location'] ?? 'Online',
      capacity: json['capacity'] ?? 0,
      categoryName: (json['category'] != null && json['category'] is Map) ? (json['category']['name'] ?? 'General') : 'General',
      organizerName: (json['organizer'] != null && json['organizer'] is Map) ? (json['organizer']['name'] ?? 'Organizer') : 'Organizer',
      price: (json['price'] ?? 0.0).toDouble(),
      registrationCount: json['registrationCount'] ?? 0,
    );
  }
}
